Sources/Sentry/SentryScreenshotIntegration.m

```diff
@@ -0,0 +1,65 @@
+#import "SentryScreenshotIntegration.h"
+#import "SentryAttachment.h"
+#import "SentryClient+Private.h"
+#import "SentryDependencyContainer.h"
+#import "SentryEvent.h"
+#import "SentryHub+Private.h"
+#import "SentryLog.h"
+#import "SentryOptions+Private.h"
+#import "SentrySDK+Private.h"
+
+#if SENTRY_HAS_UIKIT
+@implementation SentryScreenshotIntegration
+
+- (void)installWithOptions:(nonnull SentryOptions *)options
+{
+    if ([self shouldBeDisabled:options]) {
+        [options removeEnabledIntegration:NSStringFromClass([self class])];
+        return;
+    }
+
+    SentryClient *client = [SentrySDK.currentHub getClient];
+    client.attachmentProcessor = self;
+}
+
+- (NSArray<SentryAttachment *> *)processAttachments:(NSArray<SentryAttachment *> *)attachments
+                                           forEvent:(nonnull SentryEvent *)event
+{
+
+    if (event.stacktrace == nil)
```

Comment:
`m`: I think it would be better to let the caller decide when to attach a screenshot, meaning I would move this logic to the client.

Reply:
No problem. Will make this change

Tests coming next.

Reply:
So, I remember why I make this way. The `attachmentProcessor` is supposed to be a genetic thing. SentryClient doesn't need to know what is screenshot, how to get one. It just need to call `attachmentProcessor` before sending it to the transport. We may have multiple processors in the future.

If you don't think we should think ahead this way and we are good by having some logic regarding Screenshots in `SentryClient` I would rather change the name from `attachmentProcessor` to something that is related to screenshot.

Reply:
Ah okay, that makes sense. Let's keep it that way.